File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();
        int low  = 0;
        int high = n - 1;
        int insertion_index = n;
        while (low <= high){
            int mid = low + (high - low) / 2;
            if (nums[mid] >= target){
                insertion_index = mid;
                high = mid - 1;
            }
            else{
                low = mid + 1;
            }
            
        }
        return insertion_index;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n = len(nums)
        l, r = 0, n-1
        ans = n
        while l <= r:
            mid = (l+r)//2
            if nums[mid] >= target:
                ans = mid
                r = mid - 1
            else:
                l = mid + 1
        return ans